server/utils.go

```diff
@@ -176,3 +182,57 @@ func mapKeys[K comparable, V any](m map[K]V) []K {
 	}
 	return keys
 }
+
+func sanitizeFilename(name string) string {
+	return filenameSanitizationRE.ReplaceAllString(name, "_")
+}
+
+func (p *Plugin) genFilenameForCall(channelID string) (filename string) {
+	name := channelID
+	filename = fmt.Sprintf("Call_%s_%s", name, time.Now().UTC().Format("2006-01-02_15-04-05"))
+
+	channel, appErr := p.API.GetChannel(channelID)
+	if appErr != nil {
+		p.LogError("failed to get channel", "err", appErr.Error())
+		return
+	}
+
+	if channel.Type == model.ChannelTypeOpen || channel.Type == model.ChannelTypePrivate {
+		name = channel.DisplayName
+	} else if channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
+		users, appErr := p.API.GetUsersInChannel(channel.Id, model.ChannelSortByUsername, 0, 8)
+		if appErr != nil {
+			p.LogError("failed to get channel users", "err", appErr.Error())
+			return
+		}
+
+		cfg := p.API.GetConfig()
+		if cfg == nil {
+			p.LogError("failed to get configuration")
+			return
+		}
+
+		nameFormat := model.ShowUsername
+		if cfg.PrivacySettings.ShowFullName != nil && *cfg.PrivacySettings.ShowFullName {
+			nameFormat = model.ShowFullName
+		}
+
+		// We simply concatenate all the members display names separated by a dash.
+		name = ""
+		for i, u := range users {
+			name += u.GetDisplayName(nameFormat)
+			if i != len(users)-1 {
+				name += "-"
+			}
+		}
+	}
+
+	// Truncating if too long (e.g. group channels)
+	if len(name) > channelNameMaxLength {
```

Comment:
In the case where `len(name) == channelNameMaxLength + 1` you end up truncating `really long name` to `really long nam…`, and for the truncated string `len(name) == channelNameMaxLength + 3`, which is counter-intuitive, confusing, and potentially harmful.

You could also end up truncating in the middle of a rune, resulting in an invalid UTF-8 encoding in the output, which is definitely incorrect and harmful.

Reply:
Maybe something like this: https://github.com/mattermost/mattermost/blob/bb02b350481a0f111b6f0ad807de674806118a96/server/public/model/link_metadata.go#L47